protocols/kad/src/handler.rs

```diff
@@ -485,7 +484,10 @@ where
                     _ => false,
                 });
                 if let Some(pos) = pos {
-                    let _ = self.substreams.remove(pos).try_close();
+                    // TODO: we don't properly close down the substream
```

Comment:
Note that this is consistent with the current behaviour, but I realized that if the substream wasn't ready to be closed, we would just drop it immediately. That's not great.

Reply:
Closing a substream requires sending a message saying "close this substream" on the wire, and the closing process normally does that.

If a substream is dropped without being properly closed, the implementation of `StreamMuxer` can detect that and send the "close" message on the wire the next time it gets polled, but that's not an ideal way to do and I'm not sure that we're actually doing this.

Reply:
I'm ok with the current approach as long as we refactor it later with async/await.